helpers/Physics.h

```diff
@@ -0,0 +1,320 @@
+	}
+	dest->limbsLength = limbLengthDest;
+	dest->spheres.centers = spheresCenters;
+	dest->spheres.num = spheresArrayCount;
+}
+
+Z64HDR_HELPER_QUALIFIERS
+void Physics_DrawDynamicStrand(GraphicsContext* gfxCtx, TwoHeadGfxArena* disp, PhysicsLimb* jointTable, PhysicsStrand* strand, void* callback, void* callbackArg1, void* callbackArg2) {
+	s32 i;
+	f32 tempY;
+	f32 angX;
+	f32 angY;
+	Vec3f workVec;
+	Vec3f posAdd;
+	Vec3f rigidity, smoothedRigid;
+	Vec3f velAdj = { 0 };
+	Vec3f* pPos;
+	Vec3f* pRot;
+	Vec3f* pVel;
+	Vec3f* pPrevPos;
+	Vec3f* pPrevRot;
+	
+	Matrix_Push();
+	
+	jointTable[0].pos.x = strand->head.pos.x;
+	jointTable[0].pos.y = strand->head.pos.y;
+	jointTable[0].pos.z = strand->head.pos.z;
+	
+	for (i = 1; i < strand->info.numLimbs + 1; i++) {
+		Math_ApproachF(&jointTable[i].vel.x, 0.0f, 1.0f, strand->info.velStep);
+		Math_ApproachF(&jointTable[i].vel.y, 0.0f, 1.0f, strand->info.velStep);
+		Math_ApproachF(&jointTable[i].vel.z, 0.0f, 1.0f, strand->info.velStep);
+	}
+	
+	Matrix_RotateX_s(strand->head.rot.x, MTXMODE_NEW);
+	Matrix_RotateY_s(strand->head.rot.y, MTXMODE_APPLY);
+	Matrix_RotateZ_s(strand->head.rot.z, MTXMODE_APPLY);
+	if (strand->rigidity.rot.x != 0) Matrix_RotateX_f(strand->rigidity.rot.x, MTXMODE_APPLY);
+	if (strand->rigidity.rot.y != 0) Matrix_RotateY_f(strand->rigidity.rot.y, MTXMODE_APPLY);
+	if (strand->rigidity.rot.z != 0) Matrix_RotateZ_f(strand->rigidity.rot.z, MTXMODE_APPLY);
+	Matrix_MultVec3f(&strand->rigidity.push, &rigidity);
+	
+	// Main calculation loop
+	for (i = 1; i < strand->info.numLimbs + 1; i++) {
+		pPos = &jointTable[i].pos;
+		pVel = &jointTable[i].vel;
+		pPrevPos = &jointTable[i - 1].pos;
+		pPrevRot = &jointTable[i - 1].rot;
+		
+		// Smoothens curve at the start of the limb array
+		smoothedRigid = (Vec3f) { 0 };
+		if (i < strand->rigidity.num) {
+			smoothedRigid.x = ((strand->rigidity.num - i) * rigidity.x) * strand->rigidity.mult;
+			smoothedRigid.y = ((strand->rigidity.num - i) * rigidity.y) * strand->rigidity.mult;
+			smoothedRigid.z = ((strand->rigidity.num - i) * rigidity.z) * strand->rigidity.mult;
+		}
+		
+		workVec.x = pPos->x + pVel->x - pPrevPos->x + smoothedRigid.x;
+		tempY = pPos->y + pVel->y + strand->info.gravity + smoothedRigid.y;
+		workVec.z = pPos->z + pVel->z - pPrevPos->z + smoothedRigid.z;
+		
+		// FLOOR, also gets rid of the smoothedRigid
+		if (tempY < strand->info.floorY + 10.0f) {
+			workVec.x -= smoothedRigid.x;
+			workVec.z -= smoothedRigid.z;
+			if (i != strand->info.numLimbs + 1 && tempY < strand->info.floorY)
+				tempY = CLAMP_MIN(tempY, strand->info.floorY);
+		}
+		
+		workVec.y = tempY - pPrevPos->y;
+		
+		angY = Math_Atan2F(workVec.z, workVec.x);
+		angX = -Math_Atan2F(sqrtf(SQ(workVec.x) + SQ(workVec.z)), workVec.y);
+		pPrevRot->y = angY;
+		pPrevRot->x = angX;
+		
+		// Handle constraints if they are set
+		if (strand->constraint.rotStepCalc.x || strand->constraint.rotStepCalc.y || strand->constraint.lockRoot) {
```

Comment:
```suggestion
		if (strand->constraint.rotStepCalc.x != 0 || strand->constraint.rotStepCalc.y != 0 || strand->constraint.lockRoot) {
```
same with other `rotStepCalc.x`/`y` used in conditionals